addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -121,8 +123,37 @@ public void setTcsMode(String tcsId, String mode) {
         tryToCall(() -> apiClient.setTcsMode(tcsId, mode));
     }
 
-    public void setPermanentSetPoint(String zoneId, double doubleValue) {
-        tryToCall(() -> apiClient.setHeatingZoneOverride(zoneId, doubleValue));
+    public void setHeatSetPoint(String zoneId, double doubleValue) {
+        if (configuration.temporaryOverrideSchedule) {
+            try {
+                ScheduleHelper scheduleHelper = new ScheduleHelper(apiClient.requestSchedules(zoneId));
+
+                if (scheduleHelper.hasSchedule()) {
+                    ZonedDateTime zoned = ZonedDateTime.now(ZoneId.systemDefault());
+                    double scheduleTemperature = scheduleHelper.getScheduleTemperature(zoned);
+
+                    if (Double.compare(doubleValue, scheduleTemperature) == 0) {
+                        logger.debug("New temperature {} is the current scheduled temperature, cancel override for zone {}",
+                                doubleValue, zoneId);
+                        tryToCall(() -> apiClient.cancelHeatingZoneOverride(zoneId));
+                    } else {
+                        ZonedDateTime until = scheduleHelper.getStartNextSwitchPoint(zoned);
+                        logger.debug("Set temporary temperature for zone {} to {} until next switch point in schedule at {}",
+                                zoneId, doubleValue, until);
+                        tryToCall(() -> apiClient.setHeatingZoneOverride(zoneId, doubleValue, until));
+                    }
+                } else {
```

Comment:
I think I'd be nice to default to a limited timespan like midnight instead of reverting to permanent. It kind of breaks the expectation set with the 'temporary override' setting.